rocketpy/sensors/gnss.py

```diff
@@ -0,0 +1,144 @@
+            position in meters. Default is 0.
+        altitude_accuracy : float
+            Accuracy of the sensor interpreted as the standard deviation of the
+            position in meters. Default is 0.
+        name : str
+            The name of the sensor. Default is "GNSS".
+        """
+        super().__init__(sampling_rate=sampling_rate, name=name)
+        self.position_accuracy = position_accuracy
+        self.altitude_accuracy = altitude_accuracy
+
+        self.prints = _GNSSPrints(self)
+
+    def measure(self, time, **kwargs):
+        """Measure the position of the rocket in latitude, longitude and
+        altitude.
+
+        Parameters
+        ----------
+        time : float
+            Current time in seconds.
+        kwargs : dict
+            Keyword arguments dictionary containing the following keys:
+            - u : np.array
+                State vector of the rocket.
+            - u_dot : np.array
+                Derivative of the state vector of the rocket.
+            - relative_position : np.array
+                Position of the sensor relative to the rocket center of mass.
+            - environment : Environment
+                Environment object containing the atmospheric conditions.
+        """
+        u = kwargs["u"]
+        relative_position = kwargs["relative_position"]
+        lat, lon = kwargs["environment"].latitude, kwargs["environment"].longitude
+        earth_radius = kwargs["environment"].earth_radius
+
+        # Get from state u and add relative position
+        x, y, z = (Matrix.transformation(u[6:10]) @ relative_position) + Vector(u[0:3])
+        # Apply accuracy to the position
+        x = np.random.normal(x, self.position_accuracy)
+        y = np.random.normal(y, self.position_accuracy)
+        altitude = np.random.normal(z, self.altitude_accuracy)
+
+        # Convert x and y to latitude and longitude
+        lat1 = math.radians(lat)  # Launch lat point converted to radians
+        lon1 = math.radians(lon)  # Launch lon point converted to radians
+        drift = (x**2 + y**2) ** 0.5
+        bearing = (2 * math.pi - math.atan2(-x, y)) * (180 / math.pi)
+
+        # Applies the haversine equation to find final lat/lon coordinates
+        latitude = math.degrees(
+            math.asin(
+                math.sin(lat1) * math.cos(drift / earth_radius)
+                + math.cos(lat1)
+                * math.sin(drift / earth_radius)
+                * math.cos(math.radians(bearing))
+            )
+        )
+
+        # Applies the haversine equation to find final lat/lon coordinates
+        longitude = math.degrees(
+            lon1
+            + math.atan2(
+                math.sin(math.radians(bearing))
+                * math.sin(drift / earth_radius)
+                * math.cos(lat1),
+                math.cos(drift / earth_radius)
+                - math.sin(lat1) * math.sin(math.radians(latitude)),
+            )
+        )
```

Comment:
The haversine equation is being used in many parts of our code already. I think it's time for us to define it in `rocketpy.tools` and simply import it wherever it's necessary.

Reply:
There is already a `haversine` function in `tools.py` but it does not seem to do what I need here. Lets leave this for another PR

Reply:
there're `haversine` and `inverted_haversine` functions.
What you looking for is the `inverted_haversine` one.

Reply:
https://github.com/RocketPy-Team/RocketPy/pull/636/commits/a28e7bd3170076d4b4923baf9a0bdb006e1d3192